index.d.ts

```diff
@@ -2207,6 +2250,94 @@ declare namespace Eris {
     toJSON(props?: string[]): JSONCache;
   }
 
+  //Interaction
+  export class Interaction extends Base {
+    applicationID: string;
+    id: string;
+    token: string;
+    type: InteractionType | number;
+    version: number;
+  }
+
+  export class PingInteraction extends Interaction {
+    type: 1;
+    acknowledge(): Promise<void>;
+    pong(): Promise<void>;
+  }
+
+  export class CommandInteraction extends Interaction {
+    type: 2;
+    channelID: string;
+    data?: {
+      id: string;
+      name: string;
+      resolved?: {
+        users?: {[id: string]: User}
+        members?: { [id: string]: Omit<Member, "user" | "deaf" | "mute"> }
+        roles?: { [id: string]: Role }
+        channels?: { [id: string]: PartialChannel }
+      };
+      options?: InteractionDataOptions[];
+    };
+    guildID?: string;
+    member?: Member;
+    user?: User;
+    acknowledge(flags?: number): Promise<void>;
+    createFollowup(content: string | InteractionWebhookContent): Promise<Message>;
+    createMessage(content: string | InteractionContent): Promise<void>;
+    defer(flags?: number): Promise<void>;
+    deleteMessage(messageID: string): Promise<void>;
+    deleteOriginalMessage(): Promise<void>;
+    editMessage(messageID: string, content: string | MessageWebhookContent): Promise<Message>;
+    editOriginalMessage(content: string | MessageWebhookContent): Promise<Message>;
+    getOriginalMessage(): Promise<Message>
+  }
+
+  export class ComponentInteraction extends Interaction {
+    channelID: string;
+    data?: {
+      componentType?: 2 | 3;
```

Comment:
Same as last, data isn't optional

Reply:
I also don't think componentType and values can be missing on component interactions, but I don't have a source for that